// Copyright AGNTCY Contributors (https://github.com/agntcy)
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package agntcy.dir.search.v1;

import "agntcy/dir/search/v1/record_query.proto";

service SearchService {
  // List records that this peer is currently providing that match the given parameters.
  // This operation does not interact with the network.
  rpc Search(SearchRequest) returns (stream SearchResponse);
}

message SearchRequest {
  // List of queries to match against the records.
  repeated RecordQuery queries = 1;

  // Optional limit on the number of results to return.
  optional uint32 limit = 2;

  // Optional offset for pagination of results.
  optional uint32 offset = 3;
}

message SearchResponse {
  // The CID of the record that matches the search criteria.
  string record_cid = 1;
}
